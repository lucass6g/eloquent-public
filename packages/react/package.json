{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
