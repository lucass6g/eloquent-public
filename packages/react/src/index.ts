export {Switch, Case, Default} from './components/conditional-switch'