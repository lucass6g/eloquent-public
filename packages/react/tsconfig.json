{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
