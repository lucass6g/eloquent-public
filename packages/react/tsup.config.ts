import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})