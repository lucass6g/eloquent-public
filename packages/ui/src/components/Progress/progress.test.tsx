import { it, describe, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Progress } from "./";

describe("Progress Component", () => {
  it("renders with default value when no value is provided", () => {
    const component = render(<Progress data-testid="testeprogress" />);
    const progressElement = screen.getByTestId("testeprogress").children[0];
    expect(progressElement).toHaveStyle({ transform: "translateX(-100%)" });
    expect(component.baseElement).toMatchSnapshot();
  });

  it("renders with value when value is provided", () => {
    const component = render(
      <Progress value={50} data-testid="testeprogress" />
    );
    const progressElement = screen.getByTestId("testeprogress").children[0];
    expect(progressElement).toHaveStyle({ transform: "translateX(-50%)" });
    expect(component.baseElement).toMatchSnapshot();
  });
});
