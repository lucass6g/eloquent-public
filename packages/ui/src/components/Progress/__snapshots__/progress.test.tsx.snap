// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress Component > renders with default value when no value is provided 1`] = `
<body>
  <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      class="relative h-2 w-full overflow-hidden rounded-full bg-primary/20 border border-primary/40"
      data-max="100"
      data-state="indeterminate"
      data-testid="testeprogress"
      role="progressbar"
    >
      <div
        class="h-full w-full flex-1 bg-primary transition-all"
        data-max="100"
        data-state="indeterminate"
        style="transform: translateX(-100%);"
      />
    </div>
  </div>
</body>
`;

exports[`Progress Component > renders with value when value is provided 1`] = `
<body>
  <div>
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      class="relative h-2 w-full overflow-hidden rounded-full bg-primary/20 border border-primary/40"
      data-max="100"
      data-state="indeterminate"
      data-testid="testeprogress"
      role="progressbar"
    >
      <div
        class="h-full w-full flex-1 bg-primary transition-all"
        data-max="100"
        data-state="indeterminate"
        style="transform: translateX(-50%);"
      />
    </div>
  </div>
</body>
`;
